### ISSUE TITLE
Nonzero averages for "Best of 3" events in WCA Export

### ISSUE THREAD

**Issue by Ranzha** [cubing/ccm#323]
When exporting Best of 3 events, averages are calculated as -1 (DNF) for competitors with a DNF/DNS result. This value should be 0.

This is not a problem for Best of 2 or Best of 1 events.

Example:

```
  "events": [
    {
      "eventId": "333",
      "rounds": [
        {
          "roundId": "f",
          "formatId": "2",
          "results": [
            {
              "personId": 1,
              "position": 1,
              "results": [
                625,
                625
              ],
              "best": 625,
              "average": 0
            }
          ],
          "groups": []
        }
      ]
    },
    {
      "eventId": "444",
      "rounds": [
        {
          "roundId": "f",
          "formatId": "3",
          "results": [
            {
              "personId": 1,
              "position": 1,
              "results": [
                -1,
                625,
                625
              ],
              "best": 625,
              "average": -1
            }
          ],
          "groups": []
        }
      ]
    },
    {
      "eventId": "444bf",
      "rounds": [
        {
          "roundId": "f",
          "formatId": "3",
          "results": [
            {
              "personId": 1,
              "position": 1,
              "results": [
                38500,
                -1,
                38500
              ],
              "best": 38500,
              "average": -1
            }
          ],
          "groups": []
        }
      ]
    },
  ]
```

**Comment by jfly** [comment]
Quick comment here: I believe the described behavior actually *is* correct for 333bf best of 1, and that special case is probably the reason why we're calculating an average for non 333bf events.

Anyone who fixes this should definitely add a test, this stuff is tricky.

**Comment by Ranzha** [comment]
This hadn't been fixed in time for Reno Spring 2017, and Luis is noticeably annoyed that errors are repeating themselves.

**Comment by coder13** [comment]
Problem is 444bf and 555bf are getting their means calculated when they should just be 0.

**Comment by jfly** [comment]
Relevant WCA code: https://github.com/jfly/worldcubeassociation.org/blob/546eeb12deb4bc5732674a560bd5c55bf87e476c/webroot/results/includes/_check.php#L54-L61.

**Comment by jfly** [comment]
Reopening because https://github.com/cubing/ccm/commit/3cf64f418cfd0e7852cb60e716928311c5b362f2 broke a number of the `computeSolvesStatistics` tests in https://github.com/cubing/ccm/blob/ffa9cc8b949131efc4aac7f2d2517c880eed4860/imports/wca.tests.js#L28. These tests expect an average to be computed for best of 3, but that does not happen anymore. There are 2 fixes:

- change the tests to pass in 333bf and they can still expect an average to be computed
- change the tests to expect an average to *not* be computed

**Comment by jfly** [comment]
Fixed by 87d5c5f16f8e298c4ce7528e5546fbb9e1e1d958.
